fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -653,10 +657,24 @@ public void misMatchQueryShouldReturnNoResult(boolean shouldDeferFetch) throws E
 
     @Test
     public void threadedLuceneScanDoesntBreakPlannerAndSearch() throws Exception {
-        initializeFlat();
         CountingThreadFactory threadFactory = new CountingThreadFactory();
         executorService = Executors.newFixedThreadPool(10, threadFactory);
-        try (FDBRecordContext context = openContext()) {
+        initializeFlat();
+        FDBRecordContext context = null;
+        try {
+            for (int i = 0; i < 200; i++) {
+                context = openContext();
```

Comment:
This is just test code, but this introduces a memory leak. Each time a transaction is created, the previous one isn't closed before the `context` variable is changed to point to the new one.

I think you can get around this by restructuring the loop to be something like:

```java
for (int i = 0; i < 200; i++) {
    try (FDBRecordContext context = openContext()) {
        // insert a record with random data
        commit(context);
    }
}
```